src/core/metadata/qgslayermetadata.cpp

```diff
@@ -273,6 +278,460 @@ void QgsLayerMetadata::readFromLayer( const QgsMapLayer *layer )
   mLinks = layer->customProperty( QStringLiteral( "metadata/links" ) ).value<LinkList>();
 }
 
+bool QgsLayerMetadata::readMetadataXml( const QDomElement &metadataElement )
+{
+  QDomNode mnl;
+  QDomElement mne;
+
+  // set identifier
+  mnl = metadataElement.namedItem( QStringLiteral( "identifier" ) );
+  mIdentifier = mnl.toElement().text();
+
+  // set parent identifier
+  mnl = metadataElement.namedItem( QStringLiteral( "parentidentifier" ) );
+  mParentIdentifier = mnl.toElement().text();
+
+  // set language
+  mnl = metadataElement.namedItem( QStringLiteral( "language" ) );
+  mLanguage = mnl.toElement().text();
+
+  // set type
+  mnl = metadataElement.namedItem( QStringLiteral( "type" ) );
+  mType = mnl.toElement().text();
+
+  // set title
+  mnl = metadataElement.namedItem( QStringLiteral( "title" ) );
+  mTitle = mnl.toElement().text();
+
+  // set abstract
+  mnl = metadataElement.namedItem( QStringLiteral( "abstract" ) );
+  mAbstract = mnl.toElement().text();
+
+  // set keywords
+  QDomNodeList keywords = metadataElement.elementsByTagName( QStringLiteral( "keywords" ) );
+  for ( int i = 0; i < keywords.size(); i++ )
+  {
+    QStringList keywordsList;
+    mnl = keywords.at( i );
+    mne = mnl.toElement();
+
+    mKeywords.clear();
```

Comment:
This will need to be outside the loop, or you're clearing away all but the final keyword